### PR TITLE
Pluto: Componentize Mismatch Report

### PR DESCRIPTION
**Pull request by abrieff**
Moved the mismatch graphs into a component to continue the process of moving everything into classes. Functionality should be unchanged from dev

### CODE REVIEW

File: src/pluto/components/mismatch_report.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        self.v1 = v1
+        self.v2 = v2
+        self.v3 = v3
+        self.condo = condo
+        self.mapped = mapped
+
+    def __call__(self):
+        df = self.filter_by_options()
+        v1v2 = self.filter_by_version(df, f"{self.v1} - {self.v2}")
+        v2v3 = self.filter_by_version(df, f"{self.v2} - {self.v3}")
```
Comment (SashaWeinstein) [inline src/pluto/components/mismatch_report.py:20]: Can these go in the init too?

File: src/pluto/components/mismatch_report.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            "proxcode",
+            "irrlotcode",
+            "lottype",
+            "bsmtcode",
+            "yearbuilt",
+            "yearalter1",
+            "yearalter2",
+            "landmark",
+            "condono",
+        ]
```
Comment (SashaWeinstein) [inline src/pluto/components/mismatch_report.py:241]: These could all use a property decorator

**Review by SashaWeinstein — APPROVED** [review]
This works on my machine and is ready to be merged in. I have a couple small comments on what is essentially style I think, take a look and see if you want to implement

File: src/pluto/components/mismatch_report.py (hunk 1 above)
Comment (abrieff) [reply src/pluto/components/mismatch_report.py:20]: My philosophy is usually that the init should not have a ton of business logic - now that I know how this property thing works lol, my instinct to make this function smaller would be to just stick each of these in its own property function instead. might be overkill though, its got a certain kind of procedural logic atm that maybe makes it easier to read

File: src/pluto/components/mismatch_report.py (hunk 2 above)
Comment (abrieff) [reply src/pluto/components/mismatch_report.py:241]: yep will do

File: src/pluto/components/mismatch_report.py (hunk 1 above)
Comment (SashaWeinstein) [reply src/pluto/components/mismatch_report.py:20]: Ok that makes about the init. I personally feel that if "setting up" the object requires logic then that logic should go in the init, but this way is ok too. I imagined the call as a more pure "give me the content" method but I'll have plenty of time to make that tweak going forward if I want to.
What do you mean about readable "procedural" logic?

**Pull request merged**